sdk/formrecognizer/Azure.AI.FormRecognizer/src/TrainingOperation.cs

```diff
@@ -96,16 +114,41 @@ private async ValueTask<Response> UpdateStatusAsync(bool async, CancellationToke
                     ? await _serviceClient.GetCustomModelAsync(new Guid(Id), includeKeys: true, cancellationToken).ConfigureAwait(false)
                     : _serviceClient.GetCustomModel(new Guid(Id), includeKeys: true, cancellationToken);
 
-                if (update.Value.ModelInfo.Status != CustomFormModelStatus.Training)
+                _response = update.GetRawResponse();
+
+                if (update.Value.ModelInfo.Status == CustomFormModelStatus.Ready)
                 {
                     _hasCompleted = true;
                     _value = new CustomFormModel(update.Value);
                 }
+                else if (update.Value.ModelInfo.Status == CustomFormModelStatus.Invalid)
```

Comment:
If training fails, we are not letting users access the model and grab information like `RequestedOn`, `CompletedOn`, `TrainingDocuments` or `Errors`, right? The errors list can still be found in the exception message, though. Pointing it out in case we missed that, but I think we are not properly giving users the info the service has returned here.

(this argument doesn't make sense if this info is not returned in case of training failure, but I'm guessing it's not the case)

Reply:
@annelo-msft Thoughts?

Reply:
This approach follows the recommendation from our architects.
When training fails, we are communicating with our exceptions the modelId so the user can go and 1) ask for more information with `GetCustomModel` 2) delete the model as it takes space :P